RadClass/RadClass.py

```diff
@@ -0,0 +1,205 @@
+
+        NOTE: Returned data is normalized for live times (1s measurement
+        period less dead time). i.e. The returned data is a count rate at
+        each bin.
+
+        Return: 1D numpy array for integration interval
+        '''
+
+        # extract requisite data rows
+        data_matrix = self.processor.data_slice(self.datapath, rows)
+
+        # normalize by live times to produce count rate data
+        # processor.live can be indexed by appropriate timestamps but
+        # data_matrix only has indices for rows.
+        data_matrix = data_matrix / self.processor.live[rows][:, None]
+
+        # utilizes numpy architecture to sum data
+        total = np.sum(data_matrix, axis=0)
+
+        return total
+
+    def collect_rows(self):
+        '''
+        Uses the integration duration to collect all rows needed for analysis.
+        Uses numpy.arange() to enumerate row indices from start to end.
+
+        Return: list of rows for data indexing
+        '''
+
+        # collect start index from tracked timestamp
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+
+        # NOTE: this behavior is currently disabled by the if-statement in
+        # march() removing the 'or' portion with allow this to work
+        #
+        # if the final portion of the file is smaller than a full integration
+        # interval, only what is left is collected for this analysis
+        end_i = min(start_i + self.integration,
+                    len(self.processor.timestamps) - 1)
+
+        # enumerate number of rows to integrate exclusive of the endpoint
+        rows = np.arange(start_i, end_i)
+        return rows
+
+    def march(self):
+        '''
+        Advance a step using stride time to find the new working integration
+        interval. Also keeps track of the EOF state but checking for
+        out-of-index situations.
+
+        Returns: boolean for EOF
+        '''
+
+        # find the working interval starting index and advance stride
+        start_i, = np.where(self.processor.timestamps == self.working_time)
+        new_i = start_i[0] + self.stride
+
+        # stop analysis if EOF reached
+        # NOTE: stops prematurely, for windows of full integration only
+        running = True
+        if new_i >= (len(self.processor.timestamps) or
+                     (new_i + self.integration) >=
+                     len(self.processor.timestamps)):
+            running = False
+
+        if running:
+            # update working integration interval timestep
+            self.working_time = self.processor.timestamps[new_i]
+
+        return running
+
+    def iterate(self):
+        '''
+        Full iteration over the entirety of the data file. Runs
+        until EOF reached. Prints progress over the course of the analysis.
+        Only runs for a set node (datapath) with data already queued.
+        '''
+        bar = progressbar.ProgressBar(max_value=100, redirect_stdout=True)
```

Comment:
magic number `pb_max = 100` can then be reused in updating the progress bar

Reply:
I think this has to be fixed at 100 to make the progress bar a matter of percentage to completion. This is the reason for the calculation at the [line below](https://github.com/CNERG/RadClass/blob/5d0ad3d6b63fc782030b224631607878f7d86cdc/RadClass/RadClass.py#L168). Then, when using the timestamps to calculate a fraction of processed data, multiplying this by 100 results in a progress bar update value `<= max_value = 100`.